### ISSUE TITLE
[PR workflow] The label `SuppressionReviewRequired` should be split into subcategories

### ISSUE THREAD

**Issue by konrad-jamrozik** [Azure/azure-sdk-tools#6622]
There is a difference between suppression review that has to be done by ARM team vs one that has to be done by the SDK automation team. Separate labels should be introduced and appropriate guidance updated. Possibly even more than than 2 categories.

**Comment by konrad-jamrozik** [comment]
@mikekistler @rkmanda @JeffreyRichter @raych1 @heaths I chatted with @weshaggard and we are considering scrapping this mechanism completely. That is, going forward, we won't add any kind of "suppression review required" label nor demand the reviewer to add any "suppression approved" label to unblock the PR. The PRs won't be blocked even if suppressions are added.

The reasoning here is that all the relevant information is available on the PR, and the "suppression review required" was only serving as an additional, duplicate signal to make sure the PR reviewer is paying attention. We want to stop having and maintaining code & complexity that serves only as redundant means of making people pay more attention.

Going forward all suppressions made will be available for auditing via Git history, see my comments here:
- https://github.com/Azure/azure-sdk-tools/issues/4437#issuecomment-1874862792
- https://github.com/Azure/azure-sdk-tools/issues/7259#issuecomment-1874681669

**Comment by heaths** [comment]
Have you considered that those who can / should approve suppressions are a subset of those who generally review REST APIs? That's already true - by design - for breaking change reviews. Should it necessarily be different for suppressions? /cc @JeffreyRichter @mikekistler

**Comment by konrad-jamrozik** [comment]
@heats seems to me that suppressions are "sub-process" of the domain-specific review. That is - if there is a suppression review required, it is going to be as part of breaking change review, or ARM review, or SDK review. Hence the appropriate reviewers can evaluate the need to review the suppressions (if any, based on PR diff) at the time they do their domain-specific review.

**Comment by mikekistler** [comment]
My understanding -- which I'm stating here so someone can correct me if I'm mistaken -- is that suppressions and suppression review have no bearing on breaking change review/approval. I am not aware of any mechanism for suppressing breaking change errors/warnings. I _think_ that suppressions mainly effect LintDiff errors, which aren't the focus of breaking change reviews.

**Comment by raych1** [comment]
@weshaggard @konrad-jamrozik I'm curious to know if this proposed change includes the new design of the SDK breaking change review process, which now includes distinct suppression labels such as `BreakingChange-{Lang}-Suppression`?

**Comment by konrad-jamrozik** [comment]
@mikekistler @raych1 we are currently considering a new approach as part of the work on fine-grained breaking change approval reasons (#6374) and supporting persistent approvals/suppressions (#7259). One idea is to have the suppressions to be the source of truth for breaking changes approvals, not the labels.

**Comment by heaths** [comment]
> One idea is to have the Git-versioned file-based suppressions be the source of truth for breaking changes approvals

I'm intrigued. How would that scale since the any change to the file results in a new commit (version) of the file? Do you use something like `git blame` (there are other, more programmatic ways, but conceptually the same) to find when the suppression was introduced? If so, that is a very good idea indeed.

**Comment by konrad-jamrozik** [comment]
@heaths the idea is pretty much what you suggested. One idea we have for now is the following workflow:

1. A PR is submitted with a breaking change
2. PR reviewer adds some kind of label that approves the breaking change
3. As a result the relevant GitHub check gets neutralized. In addition, the automation creates a new PR with relevant suppressions file change. That new PR is a PR into the original PR, with the breaking change. The "next steps to merge" comment tell the PR author they now need to merge that suppressions-update-PR into their spec-breaking-PR to unblock their spec-breaking-PR.

Later on we will run some script to backfill all the suppressions files based on existing breaking-change-approvals-via-labels. 

Overall, there will be two ways to suppress breaking changes: via the labels, and via the suppressions file. We considered allowing doing this only via suppressions file, but we do want to retain the convenience for the PR reviewers of being able to approve a breaking change only by adding a label.

This is the current idea.

**Comment by konrad-jamrozik** [comment]
Related:
- https://github.com/Azure/azure-rest-api-specs/issues/28756
